source/agora/node/Validator.d

```diff
@@ -264,11 +264,14 @@ public class Validator : FullNode, API
     }
 
     /// GET /public_key
-    public override PublicKey getPublicKey () pure nothrow @safe
+    public override Identity getPublicKey (ulong challenge) nothrow @safe
     {
+        import std.random;
         endpoint_request_stats.increaseMetricBy!"agora_endpoint_calls_total"(
             1, "public_key", "http");
-        return this.config.validator.key_pair.address;
+        auto rand = uniform!ulong;
```

Comment:
Also when it comes to randomness, we should always prefer to use libsodium.

Reply:
Updated, thanks!